File: C++/2208. Minimum Operations to Halve Array Sum.cpp
class Solution {
public:
    int halveArray(vector<int>& n) {
        priority_queue<float>p(n.begin(),n.end());
        double s=accumulate(n.begin(),n.end(),0.0);
        double h=s/2.0;
        int c=0;
        while(h<s){
            s-=p.top()/2.0;
            p.push(p.top()/2.0);
            c++;
            if(h>s)
            return c;
            p.pop();
        }
        return c;
    }
};

File: JAVA/2208. Minimum Operations to Halve Array Sum.java
class Solution {
    public int halveArray(int[] n) {
        PriorityQueue<Double> p = new PriorityQueue<>(Collections.reverseOrder());
        double s = 0;
        for (int i : n) {
            s += i;
            p.add((double) i);
        }
        double h = s / 2.0;
        int c = 0;
        while (s > h) {
            double a = p.poll() / 2.0; 
            s -= a;
            p.add(a);
            c++; 
        }
        return c;
    }
}